tests/components/miele/test_fan.py

```diff
@@ -112,3 +161,21 @@ async def test_api_failure(
             TEST_PLATFORM, service, {ATTR_ENTITY_ID: ENTITY_ID}, blocking=True
         )
     mock_miele_client.send_action.assert_called_once()
+
+
+async def test_set_percentage(
+    hass: HomeAssistant,
+    mock_miele_client: MagicMock,
+    setup_platform: None,
+) -> None:
+    """Test handling of exception at set_percentage."""
+    mock_miele_client.send_action.side_effect = ClientResponseError("test", "Test")
+
+    with pytest.raises(HomeAssistantError):
```

Comment:
do we have translated errors yet? if so
```suggestion
    with pytest.raises(HomeAssistantError, match="....."):
```